Fix links in CONTRIBUTING.md. (#1841)
exercism/problem-specifications: CONTRIBUTING.md

This guide covers several common scenarios pertaining to **improving the language tracks**
themselves. There are other guides about contributing to other parts of the Exercism ecosystem.

* [The Command-Line Client](https://github.com/exercism/cli/blob/main/README.md)

> * [The Website](https://github.com/exercism/website/blob/main/README.md)

Reviewer: We currently don't plan on allow contributions to the website, so maybe remove this link for now?

> * [The Exercises API](https://github.com/exercism/DEPRECATED.x-api/blob/master/CONTRIBUTING.md) (used by both the command-line client and the website)

Reviewer: I think we should also remove this link, as the information seems to be quite outdated. Agree @iHiD?

Reviewer 2: Agreed

The two best ways to get help are to

* jump into the [support chat](https://gitter.im/exercism/support).

> * open a [GitHub issue](https://github.com/exercism/problem-specifications/issues).

Reviewer: I think you might have missed this bit at the top of this file:

> This guide covers several common scenarios pertaining to improving the language tracks themselves. There are other guides about contributing to other parts of the Exercism ecosystem. As such, the links are not specific to problem-specifications.

```suggestion
* open a [GitHub issue](https://github.com/exercism/exercism/issues).
```

I do agree that this is weird, and we'll likely move much of this content to our docs at some point.

## Code of Conduct

Help us keep Exercism welcoming. Please read and abide by the

> [Code of Conduct](https://github.com/exercism/problem-specifications/blob/main/CODE_OF_CONDUCT.md).

Reviewer: ```suggestion
[Code of Conduct](https://exercism.org/docs/using/legal/code-of-conduct).
```

are used to hand-craft a test suite generator, allowing us to quickly regenerate
test suites when edge cases or errors are discovered.

The generic problem descriptions live in the `exercises`
[folder](https://github.com/exercism/problem-specifications/tree/main/exercises)

> in [this repository](https://github.com/exercism/problem-specifications).

Reviewer: I think maybe be specific here?

```suggestion
in [the exercism/problem-specifications repository](https://github.com/exercism/problem-specifications).
```

are used to hand-craft a test suite generator, allowing us to quickly regenerate
test suites when edge cases or errors are discovered.

The generic problem descriptions live in the `exercises`
[folder](https://github.com/exercism/problem-specifications/tree/main/exercises)

> in this repository ([exercism/problem-specifications](https://github.com/exercism/problem-specifications)).

Reviewer: ```suggestion
in [this repository](https://github.com/exercism/problem-specifications).
```

### In exercism/<TRACK_ID>

* Do the same as when [porting an

>   exercise](https://github.com/exercism/legacy-docs/blob/main/you-can-help/implement-an-exercise-from-specification.md).

Reviewer: This link also points to a document with outdated information. It's probably a little bit better than nothing, but we should really be adding this information to our docs website :)

### Pull Request Guidelines

See the [pull request guidelines](https://github.com/exercism/legacy-docs/blob/main/contributing/pull-request-guidelines.md) in the [legacy-docs](https://github.com/exercism/legacy-docs) repository.

### Anatomy of an Exercise

See the [Practice Exercises](https://exercism.org/docs/building/tracks/practice-exercises) page for more information on how to structure an exercise.

### Track configuration file

Each language track has a configuration file called `config.json`. Its structure is defined in [this spec](https://github.com/exercism/docs/blob/main/anatomy/tracks/config-json.md). The `config.json` file's key feature is that it defines which exercises the language track has implemented.

The [configlet](https://github.com/exercism/docs/blob/main/anatomy/tracks/configlet/README.md) tool can verify (lint) a track's configuration file.

> ### Track-level Linting with Configlet

Reviewer: ```suggestion
### Track-level Linting with configlet
```

> For more details on how to have the [configlet tool](https://exercism.org/docs/building/configlet) check your track via its linting functionality, see [this guide](https://exercism.org/docs/building/configlet/lint).

Reviewer: ```suggestion
For more details on how to have the [configlet tool](https://exercism.org/docs/building/configlet) verify your track via its linting functionality, see [this guide](https://exercism.org/docs/building/configlet/lint).
```

### Git Basics

> See [Git Basics](https://github.com/exercism/legacy-docs/blob/main/contributing/git-basics.md)

Reviewer: ```suggestion
For help with Git, see [Git Basics](https://github.com/exercism/legacy-docs/blob/main/contributing/git-basics.md)
```

### Pull Request Guidelines

See the [pull request guidelines](https://github.com/exercism/legacy-docs/blob/main/contributing/pull-request-guidelines.md)
in the [legacy-docs](https://github.com/exercism/legacy-docs) repository.

> ### Exercise Versioning

Reviewer: We no longer have versioning for exercise, so this section can be removed (I'd do that in a separate PR though).

If the exercise is versioned, then the test suite will probably have a
_book-keeping_ type test at the very bottom that asserts against a value in
each language's [reference solution](#anatomy-of-an-exercise). If the change
you're making is backwards-incompatible, then please increment the version in
both the test suite and the reference solution.

Please review the details in the [README.md](https://github.com/exercism/problem-specifications#changing-tests).

### Anatomy of an Exercise

> See the [anatomy of an exercise](https://github.com/exercism/legacy-docs/tree/main/language-tracks/exercises/anatomy),
> including reference solutions, in the [legacy-docs](https://github.com/exercism/legacy-docs) repository.

Reviewer: ```suggestion
See the [Practice Exercises](https://exercism.org/docs/building/tracks/practice-exercises) page for more information on how to structure an exercise.
```

* `ignore_pattern` - A (case insensitive) regex pattern that will cause files matching it to not be served to the student by `exercism fetch`. The default value used if this key is not present is `example`
* `solution_pattern` - A (case sensitive) regex pattern that matches solution files in the track repository. Used by [configlet](https://github.com/exercism/configlet) to check for the presence of an example solution for each problem implemented by the track.  The default value used if this key is not present is `[Ee]xample`.

### Track-level Linting with Configlet

> For more details, see [this guide](https://github.com/exercism/legacy-docs/blob/main/language-tracks/configuration/configlet.md)
> in the [legacy-docs](https://github.com/exercism/legacy-docs) repository.

Reviewer: ```suggestion
For more details on how to have the [configlet tool](https://exercism.org/docs/building/configlet) check your track via its linting functionality, see [this guide](https://exercism.org/docs/building/configlet/lint).
```
